yarn-project/circuits.js/src/structs/kernel/combined_accumulated_data.ts

```diff
@@ -33,29 +33,20 @@ import { SideEffect, SideEffectLinkedToNoteHash } from '../side_effects.js';
  * Note: Not to be confused with `ContractDeploymentData`.
  */
 export class NewContractData {
-  /**
-   * Ethereum address of the portal contract on L1.
-   */
-  public portalContractAddress: EthAddress;
   constructor(
     /**
      * Aztec address of the contract.
      */
     public contractAddress: AztecAddress,
     /**
      * Ethereum address of the portal contract on L1.
-     * TODO(AD): refactor this later
-     * currently there is a kludge with circuits cpp as it emits an AztecAddress
      */
-    portalContractAddress: EthAddress | AztecAddress,
+    public portalContractAddress: EthAddress,
```

Comment:
Interesting to see here that it could pass an AztecAddress and then directly checking that it was eth address after